src/main/java/de/tum/in/www1/artemis/web/rest/AchievementResource.java

```diff
@@ -47,4 +65,52 @@ public AchievementResource(AchievementService achievementService, UserService us
         return ResponseEntity.ok(achievements);
     }
 
+    /**
+     * PUT /achievements : Updates an existing achievement.
+     *
+     * @param achievement the achievement to update
+     * @return the ResponseEntity with status 200 (OK) and with body the updated achievement, or with status 500
+     *         (Internal Server Error) if the achievement couldn't be updated
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+    @PutMapping("/achievements")
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Achievement> updateAchievement(@RequestBody Achievement achievement) throws URISyntaxException {
+        log.debug("REST request to update Achievement : {}", achievement);
+
+        Course course = achievement.getCourse();
+        User user = userService.getUserWithGroupsAndAuthorities();
+        if (!authCheckService.isAtLeastInstructorInCourse(course, user)) {
+            throw new AccessForbiddenException("You are not allowed to access this resource");
+        }
+
+        if (achievement.getId() == null) {
+            return ResponseEntity.badRequest().build();
+        }
+
+        Achievement result = achievementRepository.save(achievement);
```

Comment:
The variable name `result` is here a bit misleading. I would use `response`, `achievementResponse` or just `achievement`

Reply:
yeah, makes sense 😄